src/main/scala/com/bazaarvoice/sswf/model/StepResults.scala

```diff
@@ -0,0 +1,91 @@
+package com.bazaarvoice.sswf.model
+
+sealed abstract class StepResult(message: Option[String]) {
+  def isSuccessful: Boolean
+  def isInProgress: Boolean
+}
+
+case class Success(message: Option[String]) extends StepResult(message) {
+  def this() = this(None)
```

Comment:
no, they are just alternative constructors so you don't have to deal with Option every time you want to make one, whether from Java or Scala.